code/datums/extensions/lockable.dm

```diff
@@ -0,0 +1,218 @@
+/datum/extension/lockable
+	base_type = /datum/extension/lockable
+	expected_type = /obj/item
+	flags = EXTENSION_FLAG_IMMEDIATE
+	// Configuration variables
+	var/max_code_length = 5		// Max length of the password.
+
+	// Stateful variables
+	var/locked 			= 1		// Is it locked?
+	var/code 					// The passcode currently inputed.
+	var/l_code 			= null	// The actual code of the lock
+	var/l_set 			= 0		// Whether or not the lock has been set.
+	var/l_setshort 		= 0		// Whether or not the ability to set the lock is broken.
+	var/l_hacking 		= 0		// Whether or not the lock is being hacked.
+	var/emagged 		= 0		// What or not the lock is emagged.
```

Comment:
You are mixing boolean and numerical values here, probably should go with boolean. Compressing these toggles into a bitfield would be even better.

Reply:
Convert to bitfield in future PR. Good idea.